File: core/src/main/java/com/cangwang/core/ModuleApiManager.java
package com.cangwang.core;

import java.util.HashMap;

/**
 *
 * Created by cangwang on 2018/1/7.
 */

public class ModuleApiManager {

    static ModuleApiManager instance = new ModuleApiManager();

    HashMap<Class<?extends MBaseApi>,MBaseApi> aMap;

    public static ModuleApiManager getInstance(){
        return instance;
    }

    private ModuleApiManager(){
        aMap = new HashMap<>();
    }

    public boolean containsApi(Class<?extends MBaseApi> clazz){
        return aMap.containsKey(clazz);
    }

    public <T extends MBaseApi> T getApi(Class<T> clazz){
        return (T) aMap.get(clazz);
    }

    public void putApi(Class<?extends MBaseApi> key,MBaseApi value){
        aMap.put(key, value);
    }

    public void removeApi(Class<?extends MBaseApi> key){
        aMap.remove(key);
    }
}

File: core/src/main/java/com/cangwang/core/ModuleApiManager.kt
package com.cangwang.core

import java.util.HashMap

/**
 *
 * Created by cangwang on 2018/1/7.
 */
class ModuleApiManager private constructor() {
    var aMap: HashMap<Class<out MBaseApi?>?, MBaseApi?> = HashMap()
    fun containsApi(clazz: Class<out MBaseApi?>?): Boolean {
        return aMap.containsKey(clazz)
    }

    fun <T : MBaseApi?> getApi(clazz: Class<T>?): T? {
        return aMap[clazz] as T?
    }

    fun putApi(key: Class<out MBaseApi?>?, value: MBaseApi?) {
        aMap[key] = value
    }

    fun removeApi(key: Class<out MBaseApi?>?) {
        aMap.remove(key)
    }

    companion object {
        @kotlin.jvm.JvmStatic
        var instance = ModuleApiManager()
    }
}